src/variables.jl

```diff
@@ -1421,7 +1421,13 @@ Subject to
 function relax_integrality(model::Model)
     semicont_type = _MOICON{MOI.VariableIndex,MOI.Semicontinuous{Float64}}
     semiint_type = _MOICON{MOI.VariableIndex,MOI.Semiinteger{Float64}}
-    for v in all_variables(model)
+    # Get only the binary and integer variables, as the others don't need to be modified.
+    bin_int_constraints = vcat(
+        all_constraints(model, VariableRef, MOI.ZeroOne),
+        all_constraints(model, VariableRef, MOI.Integer),
+    )
+    bin_int_variables = VariableRef.(bin_int_constraints)
+    for v in bin_int_constraints
```

Comment:
Please continue to check for the presence of semicont and semiint variables among all variables. The current logic is nonsense because a variable can't be both binary/integer and semicont/semiint. You should be able to do this without an expensive loop by asking for the number of semicontinuous or semiinteger constraints.

Reply:
Ah I see, I misunderstood what was happening in this loop. Please take a look if it's correct now.